control-plane-operator/controllers/hostedcontrolplane/oapi/deployment_test.go

```diff
@@ -0,0 +1,235 @@
+package oapi
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Ensure certain deployment fields do not get set
+func TestReconcileOpenshiftAPIServerDeploymentNoChanges(t *testing.T) {
+
+	imageName := "oapiImage"
+	// Setup expected values that are universal
+
+	// Setup hypershift hosted control plane.
```

Comment:
do we need this comment?

Reply:
no comments needed in fact, all of them come from main.